### PR TITLE
feat(survey): Add backend for user survey

### PR DESCRIPTION
**Pull request by MelindaCardenas**
Linked to #2685 

Link to the now outdated [RFC doc](https://github.com/MelindaCardenas/datahub/blob/patch-1/docs/rfc/active/2775-user-survey.md). Should I update this and include it in this PR? I could also include it into the PR I would open for adding frontend code once that is ready. Let me know what is best! 

This PR includes...

The addition of PDLs for modeling the survey responses.
- SurveyResponseInfo.pdl
- SurveyResponseAspect.pdl
- SurveyResponseKey.pdl

The addition of a survey response snapshot to keep track the aspects associated with the survey response entity.
- SurveyResponseSnapshot.pdl
- metadata-models/src/main/pegasus/com/linkedin/metadata/snapshot/Snapshot.pdl

The addition of two tests to ensure POST and GET requests for adding a survey response and retrieving it work properly.
- smoke-test/test_e2e.py

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-models/src/main/pegasus/com/linkedin/metadata/key/SurveyResponseKey.pdl [hunk 1]
```diff
@@ -0,0 +1,14 @@
+namespace com.linkedin.metadata.key
+
+/**
+ * Key for a Chart
```
Comment (gabe-lyons) [inline metadata-models/src/main/pegasus/com/linkedin/metadata/key/SurveyResponseKey.pdl:4]:
```suggestion
 * Key for a SurveyResponse
```

File: metadata-models/src/main/pegasus/com/linkedin/feedback/SurveyResponseInfo.pdl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    /*
+     * question asked to user
+     */
+    questionText: string,
+
+    /*
+     * user's id
+     */
+    user: string,
```
Comment (gabe-lyons) [inline metadata-models/src/main/pegasus/com/linkedin/feedback/SurveyResponseInfo.pdl:30]: to indicate that an id is specifically a Urn, we usually use the `Urn` type. It's just an alias for string but is more explicit.

**Review by gabe-lyons — COMMENTED** [review]
nice! aside from the comments, looking good. Thanks @MelindaCardenas!

File: metadata-models/src/main/pegasus/com/linkedin/feedback/SurveyResponseInfo.pdl [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+@Aspect = {
+  "name": "surveyResponseInfo"
+}
+
+record SurveyResponseInfo {
+    /*
+     * date when SurveyResponseInfo was created
+     */
+    createdAt: optional Time,
```
[the remark this answers is not in the document]
Comment (MelindaCardenas) [reply metadata-models/src/main/pegasus/com/linkedin/feedback/SurveyResponseInfo.pdl:15]: I was getting an error without the 'optional' before and noticed that Time was optional in other places too (ex: [MLModelProperties](https://github.com/linkedin/datahub/blob/9f4906ee3f1e3f31aa76937677181ecb806f7b97/metadata-models/src/main/pegasus/com/linkedin/ml/metadata/MLModelProperties.pdl#L30) ) so I followed the same approach. Just tested it locally without the 'optional' and it seems to not be giving errors anymore though 😯 so I think I'll get rid of it 👍🏼

File: smoke-test/test_e2e.py [hunk 4]
```diff
@@ -1407,3 +1407,38 @@ def test_update_schemafield(frontend_session):
...
+        headers={
+            **restli_default_headers,
+        },
+    )
+    response.raise_for_status()
+    data = response.json()
+
+    assert data["value"]
+    assert data["value"]["com.linkedin.metadata.snapshot.SurveyResponseSnapshot"]
+    assert data["value"]["com.linkedin.metadata.snapshot.SurveyResponseSnapshot"]["urn"] == urn
```
[the remark this answers is not in the document]
Comment (MelindaCardenas) [reply smoke-test/test_e2e.py]: merged the two! One small thing is happening though (which might be normal but just want to make sure). Whenever an ingest post call is made for the first time, it receives a 500 error. Every other time after that initial ingest call passes though. I tried doing an ingest curl call for something else on DH and saw the same behavior. Is this something that typically happens? Should I worry about this in the test?

**Comment by shirshanka** [comment]
Closing as this PR hasn't made any progress in a while.

**Pull request closed without merging**